src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java

```diff
@@ -47,20 +64,48 @@ public Order createOrder(Order order) {
     public Order updateOrder(Order order) {
         orderRepository.findById(order.getId()).orElseThrow(() ->
                 new NotFoundException(String.format("Order with id: [%d] not found", order.getId())));
-        Location location = order.getLocation();
-        Location locationFromDB = locationRepository.findLocationByCityAndCountryAndRegion(
-                location.getCity(), location.getCountry(), location.getRegion());
-        if (locationFromDB == null) {
-            locationRepository.save(location);
-            return orderRepository.save(order);
-        }
-        order.setLocation(locationFromDB);
+
+        notifyAboutOrder(order, CONTRACT_WAS_UPDATED);
+        order.setLocation(saveLocation(order.getLocation()));
         return orderRepository.save(order);
     }
 
     @Override
     public void deleteOrder(Integer id) {
         log.debug(String.format("Delete order with id: [%d]", id));
```

Comment:
Again, use proper format for logging. 
`log.debug("Delete order with id: [{}]", id));`